SPECS/qemu/qemu.spec

```diff
@@ -1,64 +1,257 @@
-# Disable stripping - strip is trying to strip binaries (.img, .elf, etc.) built for different architectures.
+Vendor:         Microsoft Corporation
+Distribution:   Azure Linux
+# Provide a way to skip tests via rpmbuild `--without`
+# This makes it easier to skip tests in copr repos, where
+# the qemu test suite is historically flakey
+%bcond_without check
+
+# This spec is for AzLinux
+%global azl 3
+%global azl_no_ui 1
+
 %global __strip /bin/true
 %global libfdt_version 1.6.0
 %global libseccomp_version 2.4.0
 %global libusbx_version 1.0.23
-%global meson_version 0.58.2
+%global meson_version 0.61.3
 %global usbredir_version 0.7.1
-%global ipxe_version 1.20.1
+%if 0%{?azl}
+%global ipxe_version 1.21.1
+%else
+%global ipxe_version 20200823-5.git4bd064de
+%endif
+%if 0%{?azl}
 %global excluded_targets moxie-softmmu
+%endif
 %global have_memlock_limits 0
 %global need_qemu_kvm 0
+%ifarch %{ix86}
+%global kvm_package   system-x86
+# need_qemu_kvm should only ever be used by x86
+%global need_qemu_kvm 1
+%endif
 %ifarch x86_64
 %global kvm_package   system-x86
 # need_qemu_kvm should only ever be used by x86
 %global need_qemu_kvm 1
 %endif
+%ifarch %{power64}
+%global have_memlock_limits 1
+%global kvm_package   system-ppc
+%endif
+%ifarch s390x
+%global kvm_package   system-s390x
+%endif
+%ifarch armv7hl
+%global kvm_package   system-arm
+%endif
 %ifarch aarch64
 %global kvm_package   system-aarch64
 %endif
+%ifarch %{mips}
+%global kvm_package   system-mips
+%endif
+%ifarch riscv64
+%global kvm_package   system-riscv
+%endif
+
 %global modprobe_kvm_conf %{_sourcedir}/kvm.conf
-%ifarch x86_64
+%ifarch s390x
+    %global modprobe_kvm_conf %{_sourcedir}/kvm-s390x.conf
+%endif
+%ifarch %{ix86} x86_64
     %global modprobe_kvm_conf %{_sourcedir}/kvm-x86.conf
 %endif
+
 %global tools_only 0
+
+
+%global user_static 1
+%if 0%{?azl}
+%global user_static 0
+%endif
+%if 0%{?rhel}
+# EPEL/RHEL do not have required -static builddeps
 %global user_static 0
+%endif
+
+%global have_kvm 0
+%if 0%{?kvm_package:1}
 %global have_kvm 1
+%endif
+
+# Matches numactl ExcludeArch
 %global have_numactl 1
+%ifarch %{arm}
+%global have_numactl 0
+%endif
+
+%global have_spice 1
+%if 0%{?azl}
+%global have_spice 0
+%endif
+# Matches spice ExclusiveArch
+%ifnarch %{ix86} x86_64 %{arm} aarch64
 %global have_spice 0
+%endif
+%if 0%{?rhel} >= 9
+%global have_spice 0
+%endif
+
+# Matches xen ExclusiveArch
 %global have_xen 0
+%if 0%{?fedora}
+%ifarch x86_64 aarch64
+%global have_xen 1
+%endif
+%endif
+
+%global have_liburing 0
+%if 0%{?azl}
+%global have_liburing 1
+%endif
+%if 0%{?fedora}
+%ifnarch %{arm}
 %global have_liburing 1
+%endif
+%endif
+
+# 2.0 has this set
```

Comment:
why are we changing from 2.0?

Reply:
Do reduce the package dependencies. This is related to  enabling support in qemu for virgl - a virtual 3D capable graphics card for virtual machines running inside QEMU.

Reply:
i have no idea if we need it, but removing it is a breaking change we should make purposefully.

Reply:
I agree. But I doubt it will break any real usecase immediately. We can work on enabling it post 3/1.